adapters/socketio_berkeley.c

```diff
@@ -910,32 +910,18 @@ int socketio_send(CONCRETE_IO_HANDLE socket_io, const void* buffer, size_t size,
                 ssize_t send_result = send(socket_io_instance->socket, buffer, size, 0);
                 if ((send_result < 0) || ((size_t)send_result != size))
                 {
-                    if (send_result == INVALID_SOCKET)
+                    if (errno != EAGAIN)
                     {
-                        if (errno == EAGAIN) /*send says "come back later" with EAGAIN - likely the socket buffer cannot accept more data*/
-                        {
-                            /* queue data */
-                            if (add_pending_io(socket_io_instance, buffer, size, on_send_complete, callback_context) != 0)
-                            {
-                                LogError("Failure: add_pending_io failed.");
-                                result = MU_FAILURE;
-                            }
-                            else
-                            {
-                                /*do nothing*/
-                                result = 0;
-                            }
-                        }
-                        else
-                        {
-                            LogError("Failure: sending socket failed. errno=%d (%s).", errno, strerror(errno));
-                            result = MU_FAILURE;
-                        }
+                        LogError("Failure: sending socket failed. errno=%d (%s).", errno, strerror(errno));
+                        result = MU_FAILURE;
                     }
                     else
                     {
+                        /*send says "come back later" with EAGAIN - likely the socket buffer cannot accept more data*/
                         /* queue data */
```

Comment:
We should log in case it came here due to a EAGAIN (since it is still an error, even if treatable).

Reply:
Logging in this case is the same scenario as logging within DoWork. It would generate way too many statements, and not add up much to the debugging assets. 
Also, I would not suggest logging expected/handled failures.